src/nvim/tui/tui.c

```diff
@@ -862,7 +862,7 @@ static void fix_terminfo(TUIData *data)
     data->unibi_ext.enter_replace_mode = (int)unibi_add_ext_str(ut, NULL,
                                                                 "\x1b[3 q");
     data->unibi_ext.exit_insert_mode = (int)unibi_add_ext_str(ut, NULL,
-                                                              "\x1b[2 q");
+                                                              "\x1b[1 q");
```

Comment:
Yes, based on https://github.com/neovim/neovim/pull/3110#issuecomment-150296955 there wouldn't seem a reason for the work-around to _stop_ working at 0.42.5.  But do we want to applying  this work-around permanently?